pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -684,6 +686,14 @@ ByteBuf createConsumerStatsResponse(Consumer consumer, long requestId) {
 
     // complete the connect and sent newConnected command
     private void completeConnect(int clientProtoVersion, String clientVersion) {
+        if (clientMinVersionAllowed >= 0 && clientMinVersionAllowed > clientProtoVersion) {
+            log.info("[{}] client with version {} must be upgraded to {}", remoteAddress, clientProtoVersion,
+                    clientMinVersionAllowed);
+            final ByteBuf msg = Commands.newError(-1, ServerError.UnsupportedVersionError,
+                    "Upgrade version to " + clientMinVersionAllowed + " or higher");
```

Comment:
What about 'upgrade your client to a version that supports protocol version ...'
Otherwise it seems that you have to upgrade to Pulsar 20